### PR TITLE
KAFKA-5003: StreamThread should catch InvalidTopicException

### PR DESCRIPTION
**Pull request by mjsax**

### CODE REVIEW

**Comment by mjsax** [comment]
Same as #2747 but for `0.10.2`

**Comment by mjsax** [comment]
Updated. cf https://github.com/apache/kafka/pull/2747#discussion_r109112440

**Comment by enothereska** [comment]
@mjsax there are some differences in the test from trunk. E.g., there is no method shouldOnlyReadForEarliest here while it exists in trunk. Could you summarize in 1-2 sentences? Thanks.

**Comment by ijuma** [comment]
This is just missing the latest update to the trunk PR, which I just merged. Once @mjsax updates this PR, I'll merge it.

**Comment by mjsax** [comment]
@ijuma Cherry-picked the new test from `trunk`.

**Review by ijuma — APPROVED** [review]
LGTM, will merge after Jenkins completes.

**Comment by ijuma** [comment]
Merged to 0.10.2. Can you please close the PR?
